ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -1017,37 +964,56 @@ func generateNetworkLatencyTestCases(name, expectedHappyResponseBody string) []n
 			},
 			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter Sources is missing"),
 			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
-				agentState.EXPECT().GetTaskMetadata(endpointId).
-					Return(state.TaskResponse{}, nil).
-					Times(0)
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil).Times(0)
+			},
+		},
+		{
+			name:               fmt.Sprintf("%s incomplete request body 3", name),
+			expectedStatusCode: 400,
+			requestBody: map[string]interface{}{
+				"Sources": ipSources,
+			},
+			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter DelayMilliseconds is missing"),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil).Times(0)
 			},
 		},
 		{
 			name:               fmt.Sprintf("%s incomplete request body 3", name),
 			expectedStatusCode: 400,
 			requestBody: map[string]interface{}{
 				"DelayMilliseconds": delayMilliseconds,
-				"Sources":           ipSources,
+				"Sources":           []string{},
 			},
 			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter JitterMilliseconds is missing"),
 			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
-				agentState.EXPECT().GetTaskMetadata(endpointId).
-					Return(state.TaskResponse{}, nil).
-					Times(0)
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil).Times(0)
 			},
 		},
 		{
-			name:               fmt.Sprintf("%s incomplete request body 4", name),
+			name:               fmt.Sprintf("%s invalid LossPercent in the request body 1", name),
 			expectedStatusCode: 400,
 			requestBody: map[string]interface{}{
+				"DelayMilliseconds":  -1,
 				"JitterMilliseconds": jitterMilliseconds,
 				"Sources":            ipSources,
 			},
-			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter DelayMilliseconds is missing"),
+			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("json: cannot unmarshal number -1 into Go struct field NetworkLatencyRequest.DelayMilliseconds of type uint64"),
```

Comment:
Good catch! Let's have a new test for the case you find.

 I think we should have a check to make sure this is greater than 0. @mye956 @amogh09 thoughts?

Reply:
https://github.com/aws/amazon-ecs-agent/blame/cf8c7a6b65043c550533f330b10aef6d0a342214/ecs-agent/tmds/handlers/fault/v1/types/types.go#L97-L106

Reply:
non-blocking - can you add a follow up task for this?

Reply:
Confirmed offline. unit64 guarantees that the value will be non-negative. Since TC allows 0 as input, we should allow 0 as input as well. Thus, no action needed for this one.